src/main/java/io/dockstore/webservice/resources/DockerRepoResource.java

```diff
@@ -91,5 +101,59 @@ public String getRepos() {
         }
         return builder.toString();
     }
+    
+    
+    @GET
+    @Timed
+    @UnitOfWork
+    @Path("/registerContainer")
+    @ApiOperation(value = "Register a container",
+                  notes = "Register a container (public or private)",
+                  response = Container.class)
+    public Container registerContainer(@QueryParam("container_name") String name, @QueryParam("access_token") Long access_token){
+        Token token = tokenDAO.findById(access_token);
+        
+        if (token.getTokenSource().equals(TokenType.QUAY_IO.toString())) {
+            Optional<String> asString = ResourceUtilities.asString(TARGET_URL + "repository?public=false", token.getContent(), client);
+            //System.out.println(asString.get());
+            
+            JSONParser parser = new JSONParser();
```

Comment:
Jackson is already included to do the JSON -> Object conversion, you shouldn't need com.googlecode.json-simple as well. Look at the following example https://github.com/dropwizard/dropwizard/blob/master/dropwizard-example/src/main/java/com/example/helloworld/resources/PeopleResource.java

This should make this section much simpler and avoid having to manually copy fields like below.